### PR TITLE
Set backtrace on error object and check Piwik_ShouldPrintBackTraceWithMessage()

### PR DESCRIPTION
**Pull request by diosmosis**
… in controller instead of FrontController in case plugins decide to create their own safemode.

### CODE REVIEW

File: plugins/CorePluginsAdmin/Controller.php [hunk 1]
```diff
@@ -316,7 +316,9 @@ public function safemode($lastError = array())
 
             $errorMessage = $lastError['message'];
 
-            if (!empty($lastError['backtrace'])) {
+            if (!empty($lastError['backtrace'])
```
Comment (tsteur) [inline plugins/CorePluginsAdmin/Controller.php]: @diosmosis is it on purpose that this is only done when output format is not HTML? I wonder if it is supposed to be shown on the safemode page when backtrace with message is enabled? I think before it was the case

Comment (diosmosis) [reply plugins/CorePluginsAdmin/Controller.php]: AFAIK there was no backtrace to display in the code before (but I never paid that much attention to the safemode html).

Comment (diosmosis) [reply plugins/CorePluginsAdmin/Controller.php]: I can display it in the View as well.

**Comment by diosmosis** [comment]
Added to safemode view:

image

w/o exception:

image

Slight change in the second image since the backtrace contains the "in ... line ...".

**Comment by tsteur** [comment]
LGTM if tests pass

**Pull request merged**